src/net/ossl.cc

```diff
@@ -770,6 +772,8 @@ class session : public enable_shared_from_this<session>, public session_impl {
     session(session_type t, shared_ptr<tls::certificate_credentials> creds,
             std::unique_ptr<net::connected_socket_impl> sock, tls_options options = {})
       : _sock(std::move(sock))
+      , _local_address(fmt::format("{}", _sock->local_address()))
```

Comment:
nit, could use `fmt::to_string(_sock->local_address())` instead. simpler this way. and slightly more performant.